File: cpp/Medium/347_top_k_frequent_elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> frequent;
        unordered_map<int, int>sortedMap; 
        int val = 0;
        for( int i = 0; i < size(nums); i++){
            sortedMap[nums[i]]++;
        }
        for(auto s: sortedMap){
            if(s.second > val){
                val = s.second;
            }
        }
        while(k != 0){
            for(auto s: sortedMap){
                if(s.second == val){
                    frequent.push_back(s.first);
                    k--;
                }
                if( k == 0) break;
            }
            val--;
        }
        
        return frequent;
    }
};

File: python/medium/347_top_k_frequent_elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic = defaultdict(int)
        answer = []
        for i in nums:
            dic[i]+=1
        listt = list(sorted(dic.items(), key=lambda item: item[1]))
        #last k elements of the list and the first item
        for l in listt[-k:]:
            answer.append(l[0])
        return answer
